lib/AST/Decl.cpp

```diff
@@ -3956,16 +3954,20 @@ bool ClassDecl::inheritsSuperclassInitializers() {
   return Bits.ClassDecl.InheritsSuperclassInits;
 }
 
-AncestryOptions ClassDecl::checkAncestry() const {
-  // See if we've already computed this.
-  if (Bits.ClassDecl.AncestryComputed)
-    return AncestryOptions(Bits.ClassDecl.Ancestry);
 
+AncestryOptions ClassDecl::checkAncestry() const {
+  return AncestryOptions(evaluateOrDefault(getASTContext().evaluator,
```

Comment:
Can you have the request return AncestryOptions instead of AncestryFlags to avoid the coercion?

Reply:
OptionSet doesn’t have an equality operator for good reason (see the file).  I’m sure Sufficiently Magic Template Nonsense will let us do this eventually.